frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditLayoutSetForSubform/EditLayoutSet/CreateNewSubformLayoutSet/SubformDataModelSelect.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {
+  convertMetadataListToOptions,
+  findMetadataOptionByRelativeUrl,
+} from 'app-development/utils/metadataUtils';
+import type { MetadataOption } from 'app-development/types/MetadataOption';
+import classes from './SubformDataModelSelect.module.css';
+import type { DataModelMetadataJson } from 'app-shared/types/DataModelMetadata';
+import { useTranslation } from 'react-i18next';
+import { StudioNativeSelect } from '@studio/components';
+
+export interface ISubformDataModelSelectProps {
+  dataModels: DataModelMetadataJson[];
+  disabled: boolean;
+  selectedOption: MetadataOption | null;
+  setSelectedOption: (option: MetadataOption) => void;
+}
+
+export const SubformDataModelSelect = ({
+  dataModels,
+  disabled,
+  selectedOption,
+  setSelectedOption,
+}: ISubformDataModelSelectProps) => {
+  const { t } = useTranslation();
+  const options = convertMetadataListToOptions(dataModels);
+
+  function handleChange(e: React.ChangeEvent<HTMLSelectElement>) {
+    setSelectedOption(findMetadataOptionByRelativeUrl(options, e.target.value));
```

Comment:
I'd suggest storing only the value instead of the entire object:

```suggestion
    setSelectedOption(e.target.value);
```

Reply:
Here I am not sure exactly what do you mean? 🥲 you suggested  to use` `handleChange`(selectedDataModel: string) ` and at the same time use `(e)`, I tried to use just `event` but could not set  `setSelectedOption` to string because the type is `MetadataOption` that handles other properties we need like `fileName` or `repositoryRelativeUrl` let me know if i maybe misunderstand 🙂